src/Common/tests/System/Threading/Tasks/TaskTimeoutExtensions.cs

```diff
@@ -38,5 +38,47 @@ public static class TaskTimeoutExtensions
                 throw new TimeoutException();
             }
         }
+
+        public static async Task WhenAllOrAnyFailed(this Task[] tasks, int millisecondsTimeout)
+        {
+            var cts = new CancellationTokenSource();
+            Task task = tasks.WhenAllOrAnyFailed();
+            if (task == await Task.WhenAny(task, Task.Delay(millisecondsTimeout, cts.Token)))
+            {
+                cts.Cancel();
+                await task;
+            }
+            else
+            {
+                throw new TimeoutException();
```

Comment:
How about including a message that includes millisecondsTimeout here?

Reply:
Changed